bc_obps/compliance/service/penalty_calculation_service.py

```diff
@@ -74,6 +76,92 @@ def get_automatic_overdue_penalty_data(cls, compliance_report_version_id: int) -
             "data_is_fresh": refresh_result.data_is_fresh,
         }
 
+    @classmethod
+    def get_late_submission_penalty_data(cls, compliance_report_version_id: int) -> Dict[str, Any]:
+        """
+        Get late submission penalty data for a compliance obligation.
+        Returns late submission penalty data calculated using prime + 3% rate, accrued monthly.
+
+        Args:
+            compliance_report_version_id: The ID of the compliance report version
+
+        Returns:
+            Dictionary containing late submission penalty details:
+                - has_penalty: Boolean flag indicating if a penalty exists
+                - penalty_status: Status of the penalty from the obligation
+                - penalty_type: Type of penalty (LATE_SUBMISSION)
+                - penalty_charge_rate: Annual interest rate percentage
+                - penalty_amount: Total penalty amount
+                - faa_interest: FAA interest amount (currently 0.00)
+                - total_amount: Total amount including FAA interest
+                - data_is_fresh: Flag indicating if data is fresh from eLicensing API
+        """
+        refresh_result = ElicensingDataRefreshService.refresh_data_wrapper_by_compliance_report_version_id(
+            compliance_report_version_id=compliance_report_version_id
+        )
+        obligation = ComplianceObligation.objects.get(compliance_report_version_id=compliance_report_version_id)
+
+        # Check if late submission penalty exists
+        late_submission_penalty = CompliancePenalty.objects.filter(
+            compliance_obligation=obligation, penalty_type=CompliancePenalty.PenaltyType.LATE_SUBMISSION
+        ).first()
+
+        if not late_submission_penalty:
+            return {
+                "has_penalty": False,
+                "penalty_status": obligation.penalty_status,
+                "penalty_type": CompliancePenalty.PenaltyType.LATE_SUBMISSION,
+                "penalty_charge_rate": Decimal('0.00'),
+                "penalty_amount": Decimal('0.00'),
+                "faa_interest": Decimal('0.00'),
+                "total_amount": Decimal('0.00'),
+                "data_is_fresh": refresh_result.data_is_fresh,
+            }
+
+        # Get the final payment received date
+        final_payment = (
+            ElicensingPayment.objects.filter(
+                elicensing_line_item=(
+                    ElicensingLineItem.objects.get(
+                        elicensing_invoice=obligation.elicensing_invoice,
+                        line_item_type=ElicensingLineItem.LineItemType.FEE,
+                    )
+                )
+            )
+            .order_by('-received_date')
+            .first()
+        )
+
+        # Get interest rate for the penalty period, fallback to current rate
```

Comment:
What is this interest rate used for? It's possible that the interest rate is not the same from the accrual start date to the date of final payment